File: 프로그래머스/lv2/장정윤/n^2 배열 자르기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/87390
def solution(n, left, right):
    answer = []    
    for i in range(left, right+1):
        row = i // n
        col = i % n
        answer.append(max(row, col)+1)
    # 시간 초과
    # for i in range(1, n+1):
    #     for j in range(1, n+1):
    #         answer.append(max(i,j))
    
            
    return answer

    # [1,2,3,2,2,3,3,3,3]  
    # 1,1 => 1
    # 1,2 => 2
    # 1,3 => 3
    # 2,1 => 2
    # 2,2 => 2
    # 2,3 => 3
    # 3,1 => 3
    # 3,2 => 3
    # 3,3 => 3

File: 프로그래머스/lv2/이재표/n^2 배열 자르기.java
class Solution {
    public int[] solution(int n, long left, long right) {
        int[] answer = new int[(int)(right-left)+1];
        long startRow=left/n;
        long startColumn=left%n;
        
        long endRow=right/n;
        long endColumn=right%n;
        int idx=0;
        for(long i=left;i<=right;i++){
            int a=(int)(i/n);
            int b=(int)(i%n);
            if(a>=b){
                answer[idx++]=a+1;
            }else{
                answer[idx++]=b+1;
            }
        }
        return answer;
    }
}
